File: 071/71.go
package main

import "fmt"

func main() {
	var a_res, b_res int
	d_min := 1.0
	for b := 2; b <= 1000000; b++ {
		if b % 7 != 0 {
			a := 3 * b / 7
			d := 3.0 * (float64(b) - 7 * float64(a)) / (7 * float64(b))
			if d < d_min {
				d_min, a_res, b_res = d, a, b
			}
		}
	}
	fmt.Println(a_res, b_res)
}

File: 071/71.c
#include <stdio.h>

int main()
{
    /* a / b */
    
    int a, b, a_res, b_res;
    double d;
    double d_min = 1.0;
    for (b = 2; b <= 1000000; b++) {
        if (b % 7 != 0) {
            a = 3 * b / 7;
            d = (double) (3 * b - 7 * a) / (7 * b);
            if (d < d_min) {
                d_min = d;
                a_res = a;
                b_res = b;
            }
        }
    }

    /* fraction is irreducible */
    printf("a / b = %d / %d\n", a_res, b_res);

    return 0;
}   
